### ISSUE TITLE
Member names and augmented rules

### ISSUE THREAD

**Issue by codalogic** [arineng/jcr#4]
In 4.7.  Object Rules, it says "Each member rule of an object rule is evaluated in the order in which they appear in the object rule."  Initially I thought this would cause problems when combined with the (@augments) annotation because pre-existing regular expression names could prevent new augmenting names being added at the end.  However, I now think the rule is correct as it prevents extensions changing the meaning of members compared to how they would be interpreted without the extension, which seems like the correct behavior.  For example, if a ruleset goes to the effort of marking an object closed to extension:

```
my_obj { foo, bar, 0 //:any }
```

then extensions shouldn't be able to add to this.  If an augmenting rule ended up with an effective definition of:

```
my_obj { foo, bar, 0//:any, augmenting_rule }
```

then the processing rule described in 4.7 correctly keeps the object closed to extension without having to get into any clever logic to work out whether there are extension blocking rules (for which there may be a number of effective idioms, as indicated above) or not.

If that's reasonable, I suggest we add some text to 4.7 to highlight with an example that this is the approach we've adopted.

**Comment by codalogic** [comment]
Actually, rather than section 4.7, I can put this in the text about the @(augments) annotation.

**Comment by anewton1998** [comment]
I agree, though `0//:any` strikes me as wrong for closing an object to extensions. The 0 means no members will ever match, and therefore this object rule will always fail. I think `+@(reject)//:any` is what you want.

BTW, does the ABNF support `//`? I noticed my implementation doesn't work with it. Probably a bug I need to fix.

**Comment by anewton1998** [comment]
Are we willing to further refine this idea, or should we close it out?

**Comment by codalogic** [comment]
Good question.  As suggested by John, I could pull this out into a separate draft.  I could try to write it in a way that allows us to paste it into the main JCR draft if it's later thought mature enough.
